front-end/src/pages/Home/components/Search/Search.jsx

```diff
@@ -0,0 +1,43 @@
+// Search.js
+import React, { useState, useEffect } from 'react';
+import './Search.scss';
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faSearch } from '@fortawesome/free-solid-svg-icons';
+
+const Search = ({ onSearch ,setIsSearching}) => {
+  const [searchQuery, setSearchQuery] = useState('');
+  const [debouncedQuery, setDebouncedQuery] = useState(searchQuery);
+
+  useEffect(() => {
```

Comment:
We can eliminate the debouncedQuery state and directly debounce the searchQuery state.
Something like this:

```
  useEffect(() => {
    const handler = setTimeout(() => {
      onSearch(searchQuery);
    }, 500);

    return () => {
      clearTimeout(handler);
    };
  }, [searchQuery, onSearch]);
```

  In this case, only one useEffect will be needed to handle all of the logic.

Reply:
Done